sql/expression/function/greatest_least.go

```diff
@@ -0,0 +1,329 @@
+package function
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// Greatest returns the argument with the greatest numerical or string value. It allows for
+// numeric (ints anf floats) and string arguments and will return the used type
+// when all arguments are of the same type or floats if there are numerically
+// convertible strings or integers mixed with floats. When ints or floats
+// are mixed with non numerically convertible strings, those are ignored.
+type Greatest struct {
+	args []sql.Expression
```

Comment:
args should be exported, just so it can be transformed in a rule

Reply:
:+1: